openssh/auth.sls

```diff
@@ -12,7 +12,7 @@
       {%- else %}
     - user: {{ identifier }}
       {%- endif %}
-      {%- if 'present' in key and key['present'] and 'source' in key %}
+      {%- if 'source' in key %}
```

Comment:
Hi @polymeter ! Thanks for your PR!
I fear this change would break existing setups. Would you please alter your PR in a way that `present` if false overrides the presence of `source`?
Please also update `pillar.example`

Reply:
Hey,
wouldn't that overriding essentially make us go back to the previous behaviour? Using a combination of `present=False` with a given `source` is my very goal here.
If we need to support 2015.5 and earlier, okay - I just had not expected that, given its lifetime. However, I think we'll need to directly query the salt version and differentiate between the two cases then. Would that be okay with you?

Reply:
@polymeter my idea was to go with (your) new behavior, if `present` is absent, but stick to the old behaviour if `present: False`.
This way we don't need to care about versions, and don't break existing setups.

Reply:
I was thinking about something like this:

```jinja
{%- if 'source' in key and not (present in key and key['present'] == False) %}
```

Reply:
Hey, I don't mean to stretch this discussion unneccessarily, but I'd find that behaviour very counter-intuitive, thus requiring explicit documentation. We'd have a tristate-like option looking like a bool, and `source`-users could not simply switch `present` from `True` to `False` without breaking it.
So, I'd still vote for checking the salt version number, but I will leave the final decision to you and update the PR accordingly.